app/app.py

```diff
@@ -0,0 +1,50 @@
+from flask import Flask, render_template, Markup
+import os
+import markdown
+from adr import recipes
+from adr.recipe import run_recipe
+from adr.util.config import Configuration
+import adr
+
+app = Flask(__name__)
+recipe_names = []
+recipe_path = os.path.dirname(recipes.__file__)
+config_path = os.path.join(os.path.dirname(adr.__file__), 'config.yml')
+config = Configuration(config_path)
+
+
+def get_recipes():
+    """
+    Return a list of recipes located in /adr/recipes path
+    """
+    for file in os.listdir(recipe_path):
+        if file.endswith('.py') and file != '__init__.py':
+            recipe_names.append(file.rsplit('.', 1)[0])
+
+
+get_recipes()
+
+
+@app.route('/')
+def home():
+    """
+    Render a template with the list of availble recipes
+    :returns: template
+    """
+    return render_template('home.html', recipes=recipe_names)
+
+
+@app.route('/runrecipe/<recipename>')
+def runrecipe(recipename):
+    """
+    :param list recipename:the name of the selected recipe file
+    :returns: template
+    """
+    config.fmt = 'markdown'
+    result_md = run_recipe(recipename, None, config)

```

Comment:
I figured out this was the reason that I saw the failure. Since you're passing in `None` for the args value it means the ArgumentParser will look at `sys.argv` instead. But when running with flask this will be `['flask', 'run']`, which then fails because `run` isn't an expected argument for the recipes.

To fix this issue, you need to pass in an empty list instead of None. Once I made this change I was able to run the recipes from the web app. It's great!

Reply:
Thank @ahal for the git tips!, they are really helpful; however, my master was a mess, so I cloned the repo again and started with a clean version. I sent a new PR, please check :).

Reply:
To run the app, use the command `adr-app`